### PR TITLE
Deploy JavaScript adapter to Maven repository

### LINKED ISSUE #23312: Deploy JavaScript adapter to Maven repository
### Description

Having the JavaScript adapter available in Maven simplifies having all required artifacts available for builds done outside of the regular Keycloak release process, which relies on GitHub releases to store the artifacts.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by drichtarik**
This commit adds keycloak-js-adapter.tar.gz to the list of artifacts uploaded to Maven repository

Closes #23312

### CODE REVIEW

File: js/libs/keycloak-js/assembly.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    <fileSets>
+        <fileSet>
+            <directory>${project.basedir}/target</directory>
+            <outputDirectory>.</outputDirectory>
+            <includes>
+                <include>keycloak-*.tgz</include>
+            </includes>
+        </fileSet>
+    </fileSets>
+</assembly>
```
Comment (stianst) [inline js/libs/keycloak-js/assembly.xml:16]: This is not needed, we should rather just attach `keycloak-js-999.0.0-SNAPSHOT.tgz` as it is already created by `npm pack`.

The format of the tgz file is explicit to support installation with `npm` and other package managers. For example with the new format one can do `npm install keycloak-js-999.0.0-SNAPSHOT.tgz`.

File: js/libs/keycloak-js/pom.xml [hunk 2]
```diff
@@ -71,6 +71,50 @@
...
+                            <modifications>
+                                <modification>
+                                    <expression>$.version</expression>
+                                    <value>${project.version}</value>
+                                </modification>
+                            </modifications>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
```
Comment (stianst) [inline js/libs/keycloak-js/pom.xml:96]: This is not a reliable way to replace the versions in the `package.json`. We should really be using `set-version.sh` script to update the versions before creating a tag that is built.

This also takes care of setting versions in Maven files, containers, docs, and makes sure admin client/console uses the correct versions.

**Review by stianst — COMMENTED** [review]
Not sure if we really should upload `keycloak-js-adapter-999.0.0-SNAPSHOT.tar.gz` in community releases, but guess it doesn't really hurt so okay with doing that.

**Comment by stianst** [comment]
@drichtarik One question, can PNC upload non-Maven artifacts as part of the build? It doesn't really make all that much sense for us to upload a NPM package to Maven, especially not in community.

File: js/libs/keycloak-js/pom.xml (hunk 2 above)
Comment (jonkoops) [reply js/libs/keycloak-js/pom.xml:96]: Yes, this should be re-used if possible. Also because I believe the lockfile needs an update as well.

**Comment by pskopek** [comment]
closing this PR as the issue is address with newer  one https://github.com/keycloak/keycloak/pull/23428

**Pull request closed without merging**